components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -2147,6 +2147,64 @@ public static NativeArray jsFunction_getHTTPsGatewayEndpointURLs(Context cx, Scr
         return myn;
     }
 
+    public static JSONObject jsFunction_getHTTPSGatewayEndpointURLsWithType(Context cx, Scriptable thisObj,
+                                                                            Object[] args, Function funObj)
+            throws ScriptException, APIManagementException {
+
+        APIManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+        Map<String, Environment> environments = config.getApiGatewayEnvironments();
+        JSONObject json = new JSONObject();
+
+        for (Environment environment : environments.values()) {
+            if (APIConstants.GATEWAY_ENV_TYPE_HYBRID.equals(environment.getType())) {
+                json.put(APIConstants.GATEWAY_ENV_TYPE_PRODUCTION,
+                        APIStoreHostObject.getHttpsEnvironmentUrl(environment));
+                json.put(APIConstants.GATEWAY_ENV_TYPE_SANDBOX,
+                        APIStoreHostObject.getHttpsEnvironmentUrl(environment));
+                return json;
+            } else {
+                String environmentType = APIConstants.GATEWAY_ENV_TYPE_PRODUCTION.equals(environment.getType())
+                        ? "production" : "sandbox";
```

Comment:
Can't we use the same constant values (GATEWAY_ENV_TYPE_PRODUCTION, GATEWAY_ENV_TYPE_SANDBOX) here as well

Reply:
fixed